cmd/branch.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"github.com/spf13/cobra"
+	"github.com/tigrisdata/tigris-cli/client"
+	"github.com/tigrisdata/tigris-cli/config"
+	"github.com/tigrisdata/tigris-cli/util"
+)
+
+var checkout bool
+
+var branchCmd = &cobra.Command{
+	Use:   "branch",
+	Short: "Working with Tigris branches",
+}
+
+var createBranchCmd = &cobra.Command{
+	Use:   "create {branch_name}",
+	Short: "Creates Tigris branch",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		withLogin(cmd.Context(), func(ctx context.Context) error {
+			_, err := client.Get().UseDatabase(getProjectName()).CreateBranch(ctx, args[0])
+			if err != nil {
+				return util.Error(err, "create branch")
+			}
+
+			util.Infof("Branch %s created successfully", args[0])
+
+			if checkout {
+				checkoutBranch(args[0])
+			}
+
+			return nil
+		})
+	},
+}
+
+var deleteBranchCmd = &cobra.Command{
+	Use:   "delete {branch_name}",
+	Short: "Deletes Tigris branch",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		withLogin(cmd.Context(), func(ctx context.Context) error {
+			_, err := client.Get().UseDatabase(getProjectName()).DeleteBranch(ctx, args[0])
+			if err != nil {
+				return util.Error(err, "delete branch")
+			}
+
+			util.Infof("Branch %s deleted successfully", args[0])
+
+			return nil
+		})
+	},
+}
+
+func checkoutBranch(name string) {
```

Comment:
Does this config only apply to the tigris CLI?

Reply:
With this diff, yes.
Later we can extend. PR didn't update commit message:
```
Checkout is per user. Later we can add better granularity, e.g. per
project directory.
```

Reply:
We need to unify config, so all SDKs and CLI read the same set of config files, e.g. /etc/tigris.yaml, ~/.tigris.yaml ~/proj1/.tigris.yaml

Reply:
I think the current approach of reading from environment flags is much better as it works seamlessly across all the modern application deployment environments whether it is k8s, AWS lambda, Vercel, etc.

Reply:
What do you mean checkout is per-user?

Reply:
Right, environment variables should be used in production deployment.
Config files approach is convenience for local development.
User just run `tigris branch checkout` and both CLI and his compiled application use the same branch.

Reply:
/etc/tigris/tigris.yaml (system)
~/tigris/tigris-cli.yaml (per-user) (this is where branch is saved currently)
~/proj1/.tigris.yaml (per project)

Reply:
IMO there should be just one source of truth which the application reads the config from, which is what the environment file is. This is also the convention in the TypeScript world.

Reply:
When we implement per project branch handling, we can save it in the TypeScript .env file, but this is out of scope of this diff.

Reply:
Does the typescript sdk read this config file? If not then the env discussion is unrelated in the context of this diff.

Reply:
It doesn't.